docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst

```diff
@@ -0,0 +1,616 @@
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| ``...``                                                    | ForwardParameter_      |                      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| ``**nil``                                                  | HashSplatNilParameter_ |                      | Indicates that there are no keyword parameters or keyword patterns |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| ``**`` IDENTIFIER                                          | HashSplatParameter_    | NamedParameter_      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| IDENTIFIER ``:`` «Expr_»?                                  | KeywordParameter_      | NamedParameter_      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| IDENTIFIER ``=`` Expr_                                     | OptionalParameter_     | NamedParameter_      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| IDENTIFIER                                                 | SimpleParameter_       | NamedParameter_      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+| ``*`` IDENTIFIER                                           | SplatParameter_        | NamedParameter_      |                                                                    |
++------------------------------------------------------------+------------------------+----------------------+--------------------------------------------------------------------+
+
+Pattern classes
+~~~~~~~~~~~~~~~
+
+All classes in this subsection are subclasses of CasePattern_. These expressions typically occur in the context of a ``case`` using pattern matching syntax.
```

Comment:
Minor suggestion. Does this retain the correct meaning?

```suggestion
All classes in this subsection are subclasses of CasePattern_. These expressions typically occur when a ``case`` uses pattern matching syntax.
```

Reply:
Yep, this reads correctly to me and seems clearer.